diff --git a/cupsfilters/ppdgenerator.c b/cupsfilters/ppdgenerator.c
index 934b1ce..8d12193 100644
--- a/cupsfilters/ppdgenerator.c
+++ b/cupsfilters/ppdgenerator.c
@@ -1448,19 +1448,6 @@ ppdCreateFromIPP(char   *buffer,	/* I - Filename buffer */
     formatfound = 1;
     is_pdf = 1;
   }
-  if (cupsArrayFind(pdl_list, "image/pwg-raster")) {
-    if ((attr = ippFindAttribute(response, "pwg-raster-document-resolution-supported", IPP_TAG_RESOLUTION)) != NULL) {
-      current_def = NULL;
-      if ((current_res = ippResolutionListToArray(attr)) != NULL &&
-	  joinResolutionArrays(&common_res, &current_res, &common_def,
-			       &current_def)) {
-	cupsFilePuts(fp, "*cupsFilter2: \"image/pwg-raster image/pwg-raster 0 -\"\n");
-	if (formatfound == 0) manual_copies = 1;
-	formatfound = 1;
-	is_pwg = 1;
-      }
-    }
-  }
 #ifdef CUPS_RASTER_HAVE_APPLERASTER
   if (cupsArrayFind(pdl_list, "image/urf")) {
     if ((attr = ippFindAttribute(response, "urf-supported", IPP_TAG_KEYWORD)) != NULL) {
@@ -1500,6 +1487,19 @@ ppdCreateFromIPP(char   *buffer,	/* I - Filename buffer */
     }
   }
 #endif
+  if (cupsArrayFind(pdl_list, "image/pwg-raster")) {
+    if ((attr = ippFindAttribute(response, "pwg-raster-document-resolution-supported", IPP_TAG_RESOLUTION)) != NULL) {
+      current_def = NULL;
+      if ((current_res = ippResolutionListToArray(attr)) != NULL &&
+	  joinResolutionArrays(&common_res, &current_res, &common_def,
+			       &current_def)) {
+	cupsFilePuts(fp, "*cupsFilter2: \"image/pwg-raster image/pwg-raster 0 -\"\n");
+	if (formatfound == 0) manual_copies = 1;
+	formatfound = 1;
+	is_pwg = 1;
+      }
+    }
+  }
 #ifdef QPDF_HAVE_PCLM
   if (cupsArrayFind(pdl_list, "application/PCLm")) {
     if ((attr = ippFindAttribute(response, "pclm-source-resolution-supported", IPP_TAG_RESOLUTION)) != NULL) {
