diff --git a/utils/cups-browsed.c b/utils/cups-browsed.c
index 0c40112..450179b 100644
--- a/utils/cups-browsed.c
+++ b/utils/cups-browsed.c
@@ -6417,13 +6417,14 @@ on_job_state (CupsNotifier *object,
 	      }
 	      break;
 	    }
+
+	    ippDelete(response);
+	    response = NULL;
+
 	    if (pstate == IPP_PRINTER_IDLE && paccept) {
 	      q->last_printer = i;
 	      break;
 	    }
-
-	    ippDelete(response);
-            response = NULL;
 	  } else
 	    debug_printf("IPP request to %s:%d failed.\n", p->host,
 			 p->port);
