diff -up cups-filters-1.0.35/mime/cupsfilters.convs.textfilters cups-filters-1.0.35/mime/cupsfilters.convs
--- cups-filters-1.0.35/mime/cupsfilters.convs.textfilters	2013-07-29 16:08:25.244864850 +0100
+++ cups-filters-1.0.35/mime/cupsfilters.convs	2013-07-29 16:09:40.030216976 +0100
@@ -46,12 +46,12 @@ application/postscript		application/pdf
 application/vnd.adobe-reader-postscript	application/vnd.cups-postscript	66	pstops
 
 application/pdf		application/vnd.cups-pdf		66	pdftopdf
-application/x-cshell	application/pdf				0	texttopdf
-application/x-csource	application/pdf				0	texttopdf
-application/x-perl	application/pdf				0	texttopdf
-application/x-shell	application/pdf				0	texttopdf
-text/plain		application/pdf				0	texttopdf
-text/html		application/pdf				0	texttopdf
+application/x-cshell	application/pdf				200	texttopdf
+application/x-csource	application/pdf				200	texttopdf
+application/x-perl	application/pdf				200	texttopdf
+application/x-shell	application/pdf				200	texttopdf
+text/plain		application/pdf				200	texttopdf
+text/html		application/pdf				200	texttopdf
 image/gif		application/vnd.cups-pdf		33	imagetopdf
 image/png		application/vnd.cups-pdf		33	imagetopdf
 image/jpeg		application/vnd.cups-pdf		33	imagetopdf
