diff --git a/filter/pdftopdf/pdftopdf.cc b/filter/pdftopdf/pdftopdf.cc
index 7a684a260..eece5f810 100644
--- a/filter/pdftopdf/pdftopdf.cc
+++ b/filter/pdftopdf/pdftopdf.cc
@@ -745,9 +745,12 @@ bool checkFeature(const char *feature, int num_options, cups_option_t *options)
 	    // filter (PDF printer) and has to log
 	    // If output data (FINAL_CONTENT_TYPE) is PWG Raster, *toraster is
 	    // last filter (IPP Everywhere printer) and pdftopdf has to log
+	    // If output data (FINAL_CONTENT_TYPE) is PostScript, pdftops is
+	    // last filter (PostScript printer) and pdftopdf has to log
 	    if (strcasestr(final_content_type, "/pdf") ||
 		strcasestr(final_content_type, "/vnd.cups-pdf") ||
-		strcasestr(final_content_type, "/pwg-raster"))
+		strcasestr(final_content_type, "/pwg-raster") ||
+		strcasestr(final_content_type, "/postscript"))
 	      param.page_logging = 1;
 	    else
 	      param.page_logging = 0;
