diff -up cups-filters-1.0.47/Makefile.am.urftopdf cups-filters-1.0.47/Makefile.am
--- cups-filters-1.0.47/Makefile.am.urftopdf	2014-02-27 16:33:39.000000000 +0100
+++ cups-filters-1.0.47/Makefile.am	2014-03-11 12:46:24.566688998 +0100
@@ -421,8 +421,8 @@ pkgfilter_PROGRAMS += \
 	pdftoraster \
 	rastertoescpx \
 	rastertopclx \
-	texttopdf \
-	urftopdf
+	texttopdf
+
 if ENABLE_IMAGEFILTERS
 pkgfilter_PROGRAMS += \
 	imagetopdf \
diff -up cups-filters-1.0.47/Makefile.in.urftopdf cups-filters-1.0.47/Makefile.in
--- cups-filters-1.0.47/Makefile.in.urftopdf	2014-03-11 12:46:24.568688972 +0100
+++ cups-filters-1.0.47/Makefile.in	2014-03-11 12:48:31.572015731 +0100
@@ -96,7 +96,7 @@ pkgfilter_PROGRAMS = pdftoopvp$(EXEEXT)
 	commandtopclx$(EXEEXT) foomatic-rip$(EXEEXT) \
 	gstoraster$(EXEEXT) pdftoijs$(EXEEXT) pdftoippprinter$(EXEEXT) \
 	pdftops$(EXEEXT) pdftoraster$(EXEEXT) rastertoescpx$(EXEEXT) \
-	rastertopclx$(EXEEXT) texttopdf$(EXEEXT) urftopdf$(EXEEXT) \
+	rastertopclx$(EXEEXT) texttopdf$(EXEEXT) \
 	$(am__EXEEXT_1)
 @ENABLE_IMAGEFILTERS_TRUE@am__append_3 = \
 @ENABLE_IMAGEFILTERS_TRUE@	imagetopdf \
diff -up cups-filters-1.0.47/mime/cupsfilters.convs.urftopdf cups-filters-1.0.47/mime/cupsfilters.convs
--- cups-filters-1.0.47/mime/cupsfilters.convs.urftopdf	2013-12-10 22:20:15.000000000 +0100
+++ cups-filters-1.0.47/mime/cupsfilters.convs	2014-03-11 12:46:24.569688958 +0100
@@ -68,7 +68,6 @@ image/x-xpixmap		application/vnd.cups-pd
 image/x-xwindowdump	application/vnd.cups-pdf		65	imagetopdf
 image/x-sun-raster	application/vnd.cups-pdf		65	imagetopdf
 application/vnd.cups-pdf-banner	application/pdf			32	bannertopdf
-image/urf		application/pdf				0	urftopdf
 
 ########################################################################
 #
