diff -up cups-filters-1.0.40/utils/cups-browsed.c.socket-leak cups-filters-1.0.40/utils/cups-browsed.c
--- cups-filters-1.0.40/utils/cups-browsed.c.socket-leak	2013-10-02 08:53:26.000000000 +0100
+++ cups-filters-1.0.40/utils/cups-browsed.c	2013-10-21 14:03:11.907471644 +0100
@@ -1677,6 +1677,8 @@ browse_poll_cancel_subscription (browsep
 
   if (response)
     ippDelete(response);
+
+  httpClose (conn);
 }
 
 static gboolean
@@ -1787,6 +1789,9 @@ browse_poll (gpointer data)
 
 fail:
 
+  if (conn)
+    httpClose (conn);
+
   /* Call a new timeout handler so that we run again */
   g_timeout_add_seconds (BrowseInterval, browse_poll, data);
 
