diff -up cups-filters-1.0.28/configure.ac.initd cups-filters-1.0.28/configure.ac
--- cups-filters-1.0.28/configure.ac.initd	2012-12-29 14:23:28.000000000 +0100
+++ cups-filters-1.0.28/configure.ac	2013-01-02 15:50:12.513655482 +0100
@@ -287,7 +287,7 @@ elif test "x$rcdir" != xno; then
 	fi
 fi
 
-AM_CONDITIONAL([RCLINKS], [test "x$INITDDIR" = "x"])
+AM_CONDITIONAL([RCLINKS], [test "x$INITDIR" != "x"])
 
 if test "x${INITDIR}" != "x" -a "x${INITDDIR}" = "x"; then
    INITDDIR="${INITDIR}/init.d"
