diff -up cups-filters-1.16.1/utils/cups-browsed.c.location cups-filters-1.16.1/utils/cups-browsed.c
--- cups-filters-1.16.1/utils/cups-browsed.c.location	2017-10-06 13:35:52.733437212 +0200
+++ cups-filters-1.16.1/utils/cups-browsed.c	2017-10-06 13:38:11.053339580 +0200
@@ -6360,7 +6360,7 @@ static void
 browse_poll_get_printers (browsepoll_t *context, http_t *conn)
 {
   static const char * const rattrs[] = { "printer-uri-supported",
-					 "printer-info"};
+					 "printer-location", "printer-info"};
   ipp_t *request, *response = NULL;
   ipp_attribute_t *attr;
   GList *printers = NULL;
