diff --git a/utils/cups-browsed.c b/utils/cups-browsed.c
index 885bc1c..b4df0e4 100644
--- a/utils/cups-browsed.c
+++ b/utils/cups-browsed.c
@@ -8157,6 +8157,7 @@ gboolean update_cups_queues(gpointer unused) {
 	    sizes = NULL;
 	  } else {
 	    make_model = (char*)malloc(sizeof(char) * 256);
+	    printer_attributes = get_cluster_attributes(p->queue_name);
 	    if ((attr = ippFindAttribute(printer_attributes,
 					 "printer-make-and-model",
 					 IPP_TAG_TEXT)) != NULL)
@@ -8174,7 +8175,6 @@ gboolean update_cups_queues(gpointer unused) {
 	      }
 	    }
 	    default_pagesize = (char *)malloc(sizeof(char)*32);
-	    printer_attributes = get_cluster_attributes(p->queue_name);
 	    debug_printf("Generated Merged Attributes for local queue %s\n",
 			 p->queue_name);
 	    conflicts = generate_cluster_conflicts(p->queue_name,
@@ -8499,6 +8499,7 @@ gboolean update_cups_queues(gpointer unused) {
 	    sizes = NULL;
 	  } else {
 	    make_model = (char*)malloc(sizeof(char)*256);
+	    printer_attributes = get_cluster_attributes(p->queue_name);
 	    if((attr = ippFindAttribute(printer_attributes,
 					"printer-make-and-model",
 					IPP_TAG_TEXT)) != NULL)
@@ -8516,7 +8517,6 @@ gboolean update_cups_queues(gpointer unused) {
 	      }
 	    }
 	    default_pagesize = (char *)malloc(sizeof(char)*32);
-	    printer_attributes = get_cluster_attributes(p->queue_name);
 	    debug_printf("Generated Merged Attributes for local queue %s\n",
 			 p->queue_name);
 	    conflicts = generate_cluster_conflicts(p->queue_name,
