diff --git a/utils/cups-browsed.c b/utils/cups-browsed.c
index f21df6b..0d0f30f 100644
--- a/utils/cups-browsed.c
+++ b/utils/cups-browsed.c
@@ -3086,6 +3086,8 @@ void get_cluster_default_attributes(ipp_t** merged_attributes,
 				     temp->media_source, temp->media_type);
     ippSetCollection(*merged_attributes, &media_col_default, 0, current_media);
 
+    free(temp->media_source);
+    free(temp->media_type);
     free(temp);
   }
 
