diff --git a/filter/gstoraster.c b/filter/gstoraster.c
index 4aec079..f3424f3 100644
--- a/filter/gstoraster.c
+++ b/filter/gstoraster.c
@@ -895,7 +895,7 @@ main (int argc, char **argv, char *envp[])
       (t && (!strcasecmp(t, "true") || !strcasecmp(t, "on") ||
 	     !strcasecmp(t, "yes")))) {
     fprintf(stderr, "DEBUG: Ghostscript using Center-of-Pixel method to fill paths.\n");
-    cupsArrayAdd(gs_args, strdup("0 .setfilladjust"));
+    cupsArrayAdd(gs_args, strdup("0 0 .setfilladjust2"));
   } else
     fprintf(stderr, "DEBUG: Ghostscript using Any-Part-of-Pixel method to fill paths.\n");
 
