diff -up cups-filters-1.10.0/cupsfilters/colord.h.missing-ppd cups-filters-1.10.0/cupsfilters/colord.h
--- cups-filters-1.10.0/cupsfilters/colord.h.missing-ppd	2015-06-26 17:45:26.000000000 +0200
+++ cups-filters-1.10.0/cupsfilters/colord.h	2016-07-18 18:36:03.020800454 +0200
@@ -35,6 +35,7 @@ extern "C" {
 /* Common routines for accessing the colord CMS framework */
 
 #include <cups/raster.h>
+#include <cups/ppd.h>
 
 char  **colord_get_qualifier_for_ppd      (ppd_file_t *ppd);
 char   *colord_get_profile_for_device_id  (const char *device_id,
diff -up cups-filters-1.10.0/cupsfilters/colormanager.h.missing-ppd cups-filters-1.10.0/cupsfilters/colormanager.h
--- cups-filters-1.10.0/cupsfilters/colormanager.h.missing-ppd	2016-07-18 18:42:26.891818338 +0200
+++ cups-filters-1.10.0/cupsfilters/colormanager.h	2016-07-18 18:42:51.559626705 +0200
@@ -38,6 +38,7 @@ extern "C" {
 
 
 #include <cups/raster.h>
+#include <cups/ppd.h>
 
 
 
diff -up cups-filters-1.10.0/cupsfilters/driver.h.missing-ppd cups-filters-1.10.0/cupsfilters/driver.h
--- cups-filters-1.10.0/cupsfilters/driver.h.missing-ppd	2016-01-19 23:12:43.000000000 +0100
+++ cups-filters-1.10.0/cupsfilters/driver.h	2016-07-18 18:36:03.033800353 +0200
@@ -25,6 +25,7 @@ extern "C" {
 #  include <stdlib.h>
 #  include <time.h>
 #  include <math.h>
+#  include <cups/ppd.h>
 
 #  if defined(WIN32) || defined(__EMX__)
 #    include <io.h>
