diff -up cups-filters-1.0.53/filter/pstopdf.execmem cups-filters-1.0.53/filter/pstopdf
--- cups-filters-1.0.53/filter/pstopdf.execmem	2014-06-06 13:48:30.005230247 +0100
+++ cups-filters-1.0.53/filter/pstopdf	2014-06-06 13:49:05.762403657 +0100
@@ -105,7 +105,7 @@ eval "$(printf "%s" "$5" | sed -nre 's/.
 eval "$(printf "%s" "$5" | sed -nre 's/.*(^|\s)orientation-requested=4.*/orientation="${orientation:-90}"/p')"
 eval "$(printf "%s" "$5" | sed -nre 's/.*(^|\s)orientation-requested=5.*/orientation="${orientation:-270}"/p')"
 eval "$(printf "%s" "$5" | sed -nre 's/.*(^|\s)orientation-requested=6.*/orientation="${orientation:-180}"/p')"
-if printf "%s" "$5" | grep -iPq '(\s|^)landscape(=(1|on|yes|true))?(\s|$)'; then
+if printf "%s" "$5" | grep -iq '(\s|^)landscape(=(1|on|yes|true))?(\s|$)'; then
     if test -e "$PPD"; then
 	eval "$(sed -nre 's/^\*LandscapeOrientation:\s*Plus90.*/orientation="${orientation:-90}"/p' "$PPD")"
 	eval "$(sed -nre 's/^\*LandscapeOrientation:\s*Minus90.*/orientation="${orientation:-270}"/p' "$PPD")"
